content/en/blog/2019/app-identity-and-access-adapter/index.md

```diff
@@ -0,0 +1,167 @@
+
+## Installation
+
+App Identity and Access adapter can be installed using Helm directly from the github.com repository
+
+{{< text bash >}}
+$ helm repo add appidentityandaccessadapter https://raw.githubusercontent.com/ibm-cloud-security/app-identity-and-access-adapter/master/helm/appidentityandaccessadapter
+$ helm install --name appidentityandaccessadapter appidentityandaccessadapter/appidentityandaccessadapter
+{{< /text >}}
+
+Alternatively, you can clone the repo and install the Helm chart locally
+
+{{< text bash >}}
+$ git clone git@github.com:ibm-cloud-security/app-identity-and-access-adapter.git
+$ helm install ./helm/appidentityandaccessadapter --name appidentityandaccessadapter.
+{{< /text >}}
+
+## Protecting web applications
+
+Web applications are most commonly protected by the OpenID Connect (OIDC) workflow called `authorization_code`. When an unauthenticated/unauthorized user is detected, they are automatically redirected to the identity service of your choice and presented with the authentication page. When authentication completes, the browser is redirected back to an implicit `/oidc/callback` endpoint intercepted by the adapter. At this point, the adapter obtains access and identity tokens from the identity service and then redirects users back to their originally requiested URL in the web app.
+
+Authentication state and tokens are maintained by the adapter. Each request processed by the adapter will include the Authorization header bearing both access and identity tokens in the following format `Authorization: Bearer <access_token> <id_token>`
+
+Developers can read leverage the tokens for application experience adjustments, e.g. displaying user name, adjusting UI based on user role etc.
+
+In order to terminate the authenticated session and wipe tokens, aka user logout, simply redirect browser to the `/oidc/logout` endpoint under the protected service, e.g. if you're serving your app from `https://example.com/myapp`, redirect users to `https://example.com/myapp/oidc/logout`
+
+Whenever access token expires, a refresh token is used to automatically acquire new access and identity tokens without your user's needing to re-authenticate. If the configured identity provider returns a refresh token, it is persisted by the adapter and used to retrieve new access and identity tokens when the old ones expire.
+
+### Applying web application protection
+
+Protecting web applications requires creating two types of resources - use `OidcConfig` resources to define various OIDC providers, and `Policy` resources to define the web app protection policies. 
+
+```yaml
+apiVersion: "security.cloud.ibm.com/v1"
+kind: OidcConfig
+metadata:
+	name: my-oidc-provider-config
```

Comment:
I'm guessing the red markings indicate tabs. Please convert all these to spaces.